crates/ruff_linter/src/rules/refurb/rules/unnecessary_from_float.rs

```diff
@@ -293,7 +293,19 @@ fn handle_non_finite_float_special_case(
         return None;
     };
     let normalized = as_non_finite_float_string_literal(float_arg)?;
-    let replacement_text = format!(r#"{constructor_name}("{normalized}")"#);
+    let replacement_text = format!(
+        r#"{constructor_name}("{}")"#,
+        // `Decimal.from_float(float(" -nan")) == Decimal("nan")`
+        if normalized == "-nan" {
```

Comment:
The same handling is implemented in https://github.com/TaKO8Ki/ruff/blob/c3e873dd827b9ec8dd49cff723089852ce2301e1/crates/ruff_linter/src/rules/refurb/rules/verbose_decimal_constructor.rs#L165.

Is it better to embed this handling into `as_non_finite_float_string_literal` since Apart from unnecessary_from_float, the function is used only in https://github.com/TaKO8Ki/ruff/blob/c3e873dd827b9ec8dd49cff723089852ce2301e1/crates/ruff_linter/src/rules/refurb/rules/verbose_decimal_constructor.rs#L159.